src/libsystemd/meson.build

```diff
@@ -95,15 +95,38 @@ libsystemd_sources = files('''
         sd-utf8/sd-utf8.c
 '''.split()) + id128_sources + sd_daemon_c + sd_event_c + sd_login_c
 
+libsystemd_c_args = ['-fvisibility=default']
+
 libsystemd_static = static_library(
-        'systemd',
+        'systemd_static',
         libsystemd_sources,
         install : false,
         include_directories : includes,
         link_with : libbasic,
         dependencies : [threads,
                         librt],
-        c_args : ['-fvisibility=default'])
+        c_args : libsystemd_c_args)
+
+static = get_option('static-libsystemd')
+if static != 'false'
+    if static == 'true' or static == 'pic'
+        c_args = libsystemd_c_args
+        pic = true
+    else
+        c_args = libsystemd_c_args + ['-fno-PIC']
+        pic = false
+    endif
+    install_libsystemd_static = static_library(
+        'systemd',
+        libsystemd_sources,
+        include_directories : includes,
+        install : true,
+        install_dir : rootlibdir,
+        pic : pic,
+        dependencies : [threads,
+                        librt],
+        c_args : c_args)
```

Comment:
Hmm, does this mean that the meson `pic` option does not work on its own and requires flags to be twiddled manually? You don't do this in the other library, can you comment on this?

Reply:
Correct, what I've observed (using meson 0.44.1) is that if `c_args` is being set explicitly, I have to make sure it includes `-fno-PIC` for the non-PIC build to be done properly. For libudev we aren't setting `c_args`, so this isn't needed.